### ISSUE TITLE
azurectl generate region error

### ISSUE THREAD

**Issue by rjschwei** [SUSE/azurectl#152]
When using 

azurectl --region 'A_VALID_REGION' storage account create --name ... --description ... --label ... --locally-redundant

azurectl generates the following error:

ERROR: AzureStorageAccountCreateError: "AzureConfigSectionNotFound: 'Section region:US East not found in configuration file /home/rjschwei/.config/azurectl/config'"

SInce the region is specified on the command line the code should not consult the configuration file. Or at the very least do not generate a configuration error and use the region specified.

**Comment by bear454** [comment]
Right, this is a totally sensible expectation.

**Comment by bear454** [comment]
Currently we have the opposite option:

```
       azurectl setup account configure --name=<account_name> --publish-settings-file=<file>
           [--subscription-id=<subscriptionid>]
           [--region=<region_name> --storage-account-name=<storagename> --container-name=<containername> --create]
       azurectl setup account configure --name=<account_name> --management-pem-file=<file> --management-url=<url> --subscription-id=<subscriptionid>
           [--region=<region_name> --storage-account-name=<storagename> --container-name=<containername> --create]
...
    --create
        process storage and container configuration and create the
        storage account and the container in Azure
```

But you don't get the fine grained control of account creation that you get with `storage account create`.

**Comment by rjschwei** [comment]
I am not certain I get the comment about "fine grained" control. When I use the command shown in the initial description I expect a storage account to be created in Azure. With the command as shown I would expect that the credential information to access Azure is used based on the default configuration, as I am not using the --acount-name global option and then operate on the region I specified with --region.

**Comment by bear454** [comment]
By "fine grained", I meant that while `setup account configure ... --create` will create a storage account for an unconfigured region, it will not allow you to define the label, description, or backup strategy.

**Comment by rjschwei** [comment]
On 05/05/2016 02:40 PM, James Mason wrote:
> By "fine grained", I meant that while `setup account configure ... --create` will create a storage account for an unconfigured region, it will not allow you to define the label, description, or backup strategy.

I don't think that matters as 'setup account configure...' does not
actually do anything in Azure, it only operates locally to create named
references to actual manifestations in Azure.
> 
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/SUSE/azurectl/issues/152#issuecomment-217237814
> 


-- 
Robert Schweikert                   MAY THE SOURCE BE WITH YOU
Public Cloud Architect                         LINUX
rjschwei@suse.com
IRC: robjo

**Comment by bear454** [comment]
Adding the `--create` argument to `setup account configure` _does_ attempt to create the storage account and container if they don't already exist.

```
    --create
        process storage and container configuration and create the
        storage account and the container in Azure
```

**Comment by rjschwei** [comment]
On 05/05/2016 02:58 PM, James Mason wrote:
> Adding the `--create` argument to `setup account configure` _does_ attempt to create the storage account and container if they don't already exist.
> 
> ```
>     --create
>         process storage and container configuration and create the
>         storage account and the container in Azure
> ```
> 

OK, got it now. then yest that would be a consistency bug and another
issue should be opened.

> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/SUSE/azurectl/issues/152#issuecomment-217243342
> 


-- 
Robert Schweikert                   MAY THE SOURCE BE WITH YOU
Public Cloud Architect                         LINUX
rjschwei@suse.com
IRC: robjo

**Comment by bear454** [comment]
I think this bug is just fine, and I agreed right off, it's an issue we should fix, and the PR #153 is up for review ;-)
